src/nodes.coffee

```diff
@@ -500,6 +500,15 @@ exports.Root = class Root extends Base
     # end up being declared on the root block.
     o.scope.parameter name for name in o.locals or []
 
+  commentsAst: ->
+    @allComments ?=
+      for commentToken in (@allCommentTokens ? [])
```

Comment:
The `Root` node is expecting to have `@allCommentTokens` set to a list of comment tokens

So when it needs to generate AST for the comments, it first converts the comment tokens to comment node class objects and then per the usual pattern just calls `.ast()` on them